src/http/server.rs

```diff
@@ -63,20 +66,52 @@ fn create_allowed_headers(headers: &HeaderMap, allowed: &BTreeSet<String>) -> He
 
   new_headers
 }
-pub async fn start_server(config: Config) -> Result<()> {
+pub async fn start_server(config: Config, cert_path: Option<String>, key_path: Option<String>) -> Result<()> {
```

Comment:
cert_path and key_path should come from config can be grouped under server.http2 instead of cli args